utils/tail/tail.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package tail implements "tail -F" functionality following rotated logs
+package tail
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/golang/glog"
+	"golang.org/x/exp/inotify"
+)
+
+type Tail struct {
+	reader   *bufio.Reader
+	filename string
+	file     *os.File
+	stop     chan bool
+	watcher  *inotify.Watcher
+}
+
+const retryReopenInterval = time.Second
+const maxReopenAttempts = 3
+
+// NewTail starts opens the given file and watches it for deletion/rotation
+func NewTail(filename string) (t *Tail, err error) {
+	t = &Tail{
+		filename: filename,
+	}
+	t.file, err = os.Open(t.filename)
+	if err != nil {
+		glog.Errorf("Open failed on %s: %v", t.filename, err)
+		return nil, err
+	}
+	// TODO: not interested in old events?
+	//t.file.Seek(0, os.SEEK_END)
+	t.reader = bufio.NewReader(t.file)
+	t.stop = make(chan bool)
+	t.watcher, err = inotify.NewWatcher()
+	if err != nil {
+		glog.Errorf("Inotify init failed on %s: %v", t.filename, err)
+		return
```

Comment:
Just change this line to `return t, err` or `return nil, err`. Also change the return type to `(*Tail, error)` since the variable names don't add anything in that case.

IMHO there are only 2 reasons to give return types names:
1. If the return type is something like `(int, int, int, int)`, names distinguish the different ints (though I would usually prefer a struct for this)
2. If you need to modify a return value in a `defer` statement (again, this should be avoided if possible)